docs/source/topics/configfile.rst

```diff
@@ -40,11 +40,24 @@ a stage specific configuration value is needed, the ``stages`` mapping
 is checked first.  If no value is found then the top level keys will
 be checked.
 
-
 * ``api_gateway_stage`` - The name of the API gateway stage.  This
   will also be the URL prefix for your API
   (``https://endpoint/prefix/your-api``).
 
+* ``api_gateway_endpoint_type`` - The endpoint configuration of the
+  deployed API Gateway which determines how the API will be accessed,
+  can be EDGE, REGIONAL, PRIVATE. Note this value can only be set as a
+  top level key and defaults to EDGE. For more information see
+  https://amzn.to/2LofApt
+
+* ``api_gateway_endpoint_vpce`` - When configuring a Private API a VPC
```

Comment:
Do you think it makes sense to have this be a list of vpce_ids instead of a single vpce value? This is consistent with how we expose `subnet_ids` and `security_group_ids` for lambdas running in a vpc. I wonder if it is common if someone wants to have access from mutlitple vpc endpoints? Otherwise if it is not common, I'm fine just deferring to have them write out their own APIG policy.

Reply:
i think in that context it would be better to defer to the user just providing a policy, ie it feels pretty corner case then common case usage, albeit its an easy enough change to support. actually i'm a bit curious if the vpce identity carries over multiple accounts in which case i think your right it might be nicer to support it directly. i'm not sure how that would interact with the vpc dns resolution (re multi-account) setups. worth exploring.

Reply:
>> actually i'm a bit curious if the vpce identity carries over multiple accounts in which case i think your right it might be nicer to support it directly

I was too. So I spun up my private API in another account and allowed access from a VPC endpoint in another account, and I was able to make API calls from that VPC. So it definitely seems doable.

Thinking it over more, I think it is fine as is for the following reasons:

* For the Lambda security groups and subnets, we chose to use a list because that is what the API needs in setting up the Lambda function.
* If we want to support multiple vpce's in the future, we can overload the value to support a list eventually and do this if we get future requests for it.
* Once a user starts wanting to manage multiple vpce's, they will also may want to add more complicated access policies as well. At that point, it would probably make sense to just rely on the policy file.

In the end, I see this parameter as something to get a user up and running in a straightforward fashion with a private API. Then if they want to expand on the policy itself, they can grab the policy, save it in their chalice project, and update it as needed.

Reply:
switched over to supporting a list or string of vpce ids.